### ISSUE TITLE
Crash report - Error starting up application System.IO.FileFormatException: Formato de imagen no reconocido

### ISSUE THREAD

**Issue by JuliusSweetland** [JuliusSweetland/OptiKey#107]
````
2015-09-17 02:33:07,437 [ 1] ERROR JuliusSweetland.OptiKey.App: Error starting up application
System.IO.FileFormatException: Formato de imagen no reconocido. ---&gt; System.Runtime.InteropServices.COMException: Excepción de HRESULT: 0x88982F07
   --- Fin del seguimiento de la pila de la excepción interna ---
   en System.Windows.Media.Imaging.BitmapFrameDecode.EnsureThumbnail()
   en System.Windows.Media.Imaging.BitmapFrameDecode.get_Thumbnail()
   en MS.Internal.AppModel.IconHelper.GetBestMatch(ReadOnlyCollection`1 frames, Size size)
   en MS.Internal.AppModel.IconHelper.CreateIconHandleFromImageSource(ImageSource image, Size size)
   en MS.Internal.AppModel.IconHelper.GetIconHandlesFromImageSource(ImageSource image, IconHandle&amp; largeIconHandle, IconHandle&amp; smallIconHandle)
   en System.Windows.Window.UpdateIcon()
   en System.Windows.Window.SetupInitialState(Double requestedTop, Double requestedLeft, Double requestedWidth, Double requestedHeight)
   en System.Windows.Window.CreateSourceWindow(Boolean duringShow)
   en System.Windows.Interop.WindowInteropHelper.EnsureHandle()
   en JuliusSweetland.OptiKey.Services.WindowManipulationService..ctor(Window window, Func`1 getOpacity, Func`1 getWindowState, Func`1 getPreviousWindowState, Func`1 getFloatingSizeAndPosition, Func`1 getDockPosition, Func`1 getDockSize, Func`1 getFullDockThicknessAsPercentageOfScreen, Func`1 getCollapsedDockThicknessAsPercentageOfFullDockThickness, Func`1 getMinimisedPosition, Action`1 saveOpacity, Action`1 saveWindowState, Action`1 savePreviousWindowState, Action`1 saveFloatingSizeAndPosition, Action`1 saveDockPosition, Action`1 saveDockSize, Action`1 saveFullDockThicknessAsPercentageOfScreen, Action`1 saveCollapsedDockThicknessAsPercentageOfFullDockThickness) en c:\Users\Julius\Documents\GitHub\OptiKey\src\Services\WindowManipulationService.cs:línea 102
   en JuliusSweetland.OptiKey.App.&lt;App_OnStartup&gt;d__34.MoveNext() en c:\Users\Julius\Documents\GitHub\OptiKey\src\App.xaml.cs:línea 147
2015-09-17 02:33:07,715 [ 1] ERROR JuliusSweetland.OptiKey.App: A DispatcherUnhandledException has been encountered...
System.IO.FileFormatException: Formato de imagen no reconocido. ---&gt; System.Runtime.InteropServices.COMException: Excepción de HRESULT: 0x88982F07
   --- Fin del seguimiento de la pila de la excepción interna ---
   en System.Windows.Media.Imaging.BitmapFrameDecode.EnsureThumbnail()
   en System.Windows.Media.Imaging.BitmapFrameDecode.get_Thumbnail()
   en MS.Internal.AppModel.IconHelper.GetBestMatch(ReadOnlyCollection`1 frames, Size size)
   en MS.Internal.AppModel.IconHelper.CreateIconHandleFromImageSource(ImageSource image, Size size)
   en MS.Internal.AppModel.IconHelper.GetIconHandlesFromImageSource(ImageSource image, IconHandle&amp; largeIconHandle, IconHandle&amp; smallIconHandle)
   en System.Windows.Window.UpdateIcon()
   en System.Windows.Window.SetupInitialState(Double requestedTop, Double requestedLeft, Double requestedWidth, Double requestedHeight)
   en System.Windows.Window.CreateSourceWindow(Boolean duringShow)
   en System.Windows.Interop.WindowInteropHelper.EnsureHandle()
   en JuliusSweetland.OptiKey.Services.WindowManipulationService..ctor(Window window, Func`1 getOpacity, Func`1 getWindowState, Func`1 getPreviousWindowState, Func`1 getFloatingSizeAndPosition, Func`1 getDockPosition, Func`1 getDockSize, Func`1 getFullDockThicknessAsPercentageOfScreen, Func`1 getCollapsedDockThicknessAsPercentageOfFullDockThickness, Func`1 getMinimisedPosition, Action`1 saveOpacity, Action`1 saveWindowState, Action`1 savePreviousWindowState, Action`1 saveFloatingSizeAndPosition, Action`1 saveDockPosition, Action`1 saveDockSize, Action`1 saveFullDockThicknessAsPercentageOfScreen, Action`1 saveCollapsedDockThicknessAsPercentageOfFullDockThickness) en c:\Users\Julius\Documents\GitHub\OptiKey\src\Services\WindowManipulationService.cs:línea 102
   en JuliusSweetland.OptiKey.App.&lt;App_OnStartup&gt;d__34.MoveNext() en c:\Users\Julius\Documents\GitHub\OptiKey\src\App.xaml.cs:línea 217
--- Fin del seguimiento de la pila de la ubicación anterior donde se produjo la excepción ---
   en System.Runtime.CompilerServices.AsyncMethodBuilderCore.&lt;ThrowAsync&gt;b__4(Object state)
   en System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   en MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
````

**Comment by evanschulte** [comment]
From what I can gather, as I cannot reproduce, is that this occurs on windows XP machines when trying to load .ico files.  Here is a good thread on the issue.

https://social.msdn.microsoft.com/Forums/vstudio/en-US/9e723d51-0e26-4609-8fe1-5e0edcf7385b/setting-windowicon-to-vista-256x256-png-icon-causes-a-fileformatexception?forum=wpf

It appears that removing the compression on the 256x256 icon may fix the issue.  Another is also reporting that they were running windows in 16bit, which of course is would be a bit trickier to handle.  

I removed the compression from the ico files, but unfortunately I don't have a machine to test this fix on :(.  Not sure if you do?

**Comment by JuliusSweetland** [comment]
@evanschulte - excellent, thank you. Could you supply (pull request?) your attempt and I'll try to find a machine to try it on. Is the crash reproduceable, i.e. could I cause the crash, merge your change and try again and then potentially see it fixed immediately?

**Comment by JuliusSweetland** [comment]
Thinking about this more I am happy to ignore the problem if they are
running 16 bit. It's not going to be common enough to bother supporting.

Regarding the icon compression here are two methods (this is to remind me
later);

1.In Axialis Icon Studio you can right click the icon and disable the
PNG-compression.
2.In IcoFx uncheck this: Tools->Preferences->Tools->Compress the images
256x256 for Windows Vista

**Comment by evanschulte** [comment]
I would tend to agree with that.  If they are running 16 bit, I imagine they may end up having more problems than just an icon anyhow.  Just a note, the latter (IcoFx) is what I used and it works well.

**Comment by JuliusSweetland** [comment]
I've used IcoFx on three computers, but ask my trials have finished now. Do
you have a new version of the ico file I could use?

**Comment by evanschulte** [comment]
Yes sir I do.  Here is a link to my gdrive folder which contains the
updated icons.

https://drive.google.com/folderview?id=0B2dSDxDo57bQTkR3dW9MQTdhU0U&usp=sharing

**Comment by JuliusSweetland** [comment]
@evanschulte - thank you. I'll use the new icons in the next release, but I
have realised that this problem isn't really worth fixing anyway as OptiKey
requires .Net 4.5 which is only available from Vista SP2 onwards (i.e. not
XP anyway)!

The other thing I've noticed is that git only thinks the management.ico
file has changed. I would have expected both to have changed?

**Comment by JuliusSweetland** [comment]
Fixed in v2.0.10

**Comment by evanschulte** [comment]
@JuliusSweetland

Sorry for the delay in response, had a busy weekend.  Anyhow, I was having
a hard time getting git to recognize the change with Main.ico for some
reason. I tried renaming, deleting and re-adding, and a few other things
with no luck.  I was running a bit short on time so I sent that google
drive link instead of continuing to mess with git.  Still a bit new to
github, sorry about that!

**Comment by JuliusSweetland** [comment]
@evanschulte not at all - I appreciate your effort. I guess the file hasn't actually changed as git looks at the file's SHA (hash code). If it's the same it's almost impossible for the file to be different. Potentially I compressed one and not the other. Either way your new icon (management.ico) is included in the new release v2.0.10, so thank you again.
